Firestore/Swift/Source/Codable/DocumentID.swift

```diff
@@ -77,13 +77,13 @@ internal protocol DocumentIDProtocol {
 @propertyWrapper
 public struct DocumentID<Value: DocumentIDWrappable & Codable>:
   DocumentIDProtocol, Codable {
-  var value: Value?
+  public private(set) var value: Value?
 
   public init(wrappedValue value: Value?) {
     self.value = value
   }
 
-  public var wrappedValue: Value? {
+  public private(set) var wrappedValue: Value? {
```

Comment:
This looks confusing - have you check if it's possible instead to only mark the setter private?

Reply:
Unfortunately I couldn't find any other syntax for lowering the access level of a setter in the Swift [Access Control](https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html) documentation or elsewhere.

Note: this has now changed slightly from your question, it's now `internal(set)`, but I agree that it looks a little confusing.